File: arrays/136_single-number/main.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int singleNumber(vector<int> &nums) {
    int ans = 0;
    for (int n : nums) {
      ans ^= n;
    }
    return ans;
  }
};

int main() {
  vector<int> nums = {1, 2, 3, 2, 3};
  cout << Solution().singleNumber(nums) << endl;
  return 0;
}

File: arrays/136_single-number/main.go
package main

import "fmt"

func singleNumber(nums []int) int {
	var ans int
	for _, n := range nums {
		ans ^= n
	}
	return ans
}

func main() {
	a := []int{4, 2, 3, 2, 3, 4, 5, 6, 6}
	fmt.Println("Single number from", a, "is", singleNumber(a))
}
